lib/the_little_thinkers_space/content.ex

```diff
@@ -83,8 +101,9 @@ defmodule TheLittleThinkersSpace.Content do
       {:error, %Ecto.Changeset{}}
 
   """
-  def delete_upload(%Upload{} = upload) do
-    ImageCacher.delete_from_cache(upload.id)
+  def delete_upload(%Upload{id: id, path: path} = upload) do
+    ImageCacher.delete_from_cache(id)
+    delete_upload_file(path)
     Repo.delete(upload)
   end
```

Comment:
There are a bunch of opportunities for things to fail here. Maybe we want to use a with statement?